lib/dbhelper.py

```diff
@@ -0,0 +1,55 @@
+from tornado import gen
+import rethinkdb as r
+import logging
+
+from .config import CONFIG
+
+
+r.set_loop_type("tornado")
+FORMAT = '[%(levelname)1.1s %(asctime)s %(name)s:%(lineno)d] %(message)s'
+
+
+class RethinkDB(object):
+    def __init__(self, dbname, tables):
+        self._connection = None
+        self.dbname = dbname
+        self.tables = tables
+        logging.basicConfig(format=FORMAT)
+        self.logger = logging.getLogger("rethinkdb." + dbname)
+
+    @gen.coroutine
+    def connection(self):
+        # loop makes sure that the connection object has been completely
+        # initialized before we use it (note that we only set the
+        # self._connection object at the END of self.init()
+        while self._connection is None:
+            yield gen.sleep(0)
+        conn = yield self._connection
+        return conn
+
```

Comment:
Also, I'm simplifying this code a little bit right now so that the function to get the global reference is taken care of for you